gson/src/main/java/com/google/gson/Gson.java

```diff
@@ -705,7 +705,7 @@ public void toJson(Object src, Type typeOfSrc, JsonWriter writer) throws JsonIOE
     } catch (IOException e) {
       throw new JsonIOException(e);
     } catch (AssertionError e) {
-      throw new AssertionError("AssertionError (GSON " + GsonBuildConfig.VERSION + "): " + e.getMessage(), e);
+      throw new AssertionError("AssertionError (GSON " + GsonBuildConfig.VERSION + "): " + e.getMessage());
```

Comment:
why are you changing these?

Reply:
Because that constructor was only introduced in 1.7, which would make it not compatible with 1.6.

Reply:
Ah! Thanks.

Reply:
Store the exception in a local and call `initCause(e)` so as not to break the chain of exceptions before throwing.

Reply:
Isn't that beyond the scope of this pull request?

Reply:
Nope. You're already changing these so I would either leave them as-is to retain the cause or make them correct for Java 6 while retaining the cause.

Reply:
OK, agreed. If I revert this change, the build will fail. I do not completely understand what exactly you expect me to do using `initCause`. Could you spell it out for me so I can commit the change?

Reply:
I think this is what Jake meant:
```
AssertionError error = new AssertionError("AssertionError (GSON " + GsonBuildConfig.VERSION + "): " + e.getMessage());
error.initCause(e); 
throw error;
```

Reply:
Precisely! Just so we don't actually lose the causal chain.

Reply:
Ahh, got it, thanks! I've fixed it and pushed those changes.